demos/device_defender_for_aws/report_builder.c

```diff
@@ -482,9 +641,48 @@ ReportBuilderStatus_t GenerateJsonReport( char * pBuffer,
         }
     }
 
+    /* Write task ids array. */
+    if( status == ReportBuilderSuccess )
+    {
+        status = writeTaskIdsArray( pCurrentWritePos,
+                                    remainingBufferLength,
+                                    pMetrics->pTaskIdsArray,
+                                    pMetrics->taskIdsArrayLength,
+                                    &( bufferWritten ) );
+
+        if( status == ReportBuilderSuccess )
+        {
+            pCurrentWritePos += bufferWritten;
+            remainingBufferLength -= bufferWritten;
+        }
+        else
+        {
+            LogError( ( "Failed to write task ids array." ) );
+        }
+    }
+
+    /* Write part5. */
+    if( status == ReportBuilderSuccess )
+    {
+        charactersWritten = snprintf( pCurrentWritePos,
+                                      remainingBufferLength,
+                                      JSON_REPORT_FORMAT_PART5 );
+
+        if( !SNPRINTF_SUCCESS( charactersWritten, remainingBufferLength ) )
+        {
+            LogError( ( "Failed to write part 5." ) );
```

Comment:
Just seeing "part 5" in log messages will be unclear to the reader. 
```suggestion
            LogError( ( "Failed to write part 5 of JSON report to buffer." ) );
```